### DISCUSSION TITLE
Add "anchor following" feature

### DISCUSSION THREAD

**Discussion by justin5267** [withastro/starlight#947] [Feature Requests]
### What version of `starlight` are you using?

0.5.3

### What is your idea?

It would be convenient if the current anchor point automatically scrolled to the center of the screen.

Thanks for reading!

### Why is this feature necessary?

In some lengthy page, it can be challenging when the current anchor point moves outside the visible area of the screen. 

### Do you have examples of this feature in other projects?

Example: the "Anchor following" feature offered by Materials for MkDocs.

https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#anchor-following

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Thanks for sharing! Do you have an example of a page using MkDocs Material’s Anchor Following feature? Would love to see what it looks like.

**Comment by justin5267** [comment]
You can refer to the official documentation of MkDocs Material for this feature. On the specific page below, if you adjust the window size so that its height becomes smaller than the sidebar length, you'll notice that scrolling with the mouse will cause the current anchor point to automatically scroll along with the document content.

[https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#anchor-following](url)
